### PR TITLE
Feat: Added Cart functionality

### PR DESCRIPTION
**Pull request by scjuly19**
- Add, remove and update cart functionality added

### CODE REVIEW

**Review by surajmore98 — APPROVED** [review]
looks all good.

File: src/components/Header.js [hunk 1]
```diff
@@ -8,11 +8,12 @@ import { RequireAuth } from "./RequireAuth";
 import { useAuthContext } from "../context/authContext/authContext";
 export default function Header() {
   const { state: dataState } = useDataContext();
-  const { authed } = useAuthContext();
-  const { loading } = dataState;
+  const { authed,loader } = useAuthContext();
+  const { loading,cartData } = dataState;
+  let cartItemsCount=cartData.length
```
Comment (rutvikpumak) [inline src/components/Header.js:13]: use const

File: src/network/dataRequests.js [hunk 2]
```diff
@@ -1,5 +1,45 @@
...
+    GET_CART_ITEMS,
+    { product },
+    {
+      headers: {
+        authorization: encodedToken,
+      },
+    }
+  );
+const quantityUpdate = async (productId, encodedToken,actionType) =>{
+  let url=GET_CART_ITEMS + "/" + productId
```
Comment (rutvikpumak) [inline src/network/dataRequests.js:22]: better if u use template literal and const here

File: src/pages/Cart.js [hunk 3]
```diff
@@ -84,28 +90,26 @@ const CartTotal = ({ cartTotal }) => {
...
+export function Cart() {
+  const { state, dispatch } = useDataContext();
+  const { token } = useAuthContext();
+  const { cartData } = state;
+  const cartTotal =
+    cartData.length > 0 &&
+    cartData.reduce((acc, item) => acc + item.price * item.qty, 0);
+
+  const handleDecrement = (id,qty) => {
+    if(qty>1){
```
Comment (rutvikpumak) [inline src/pages/Cart.js]: use ternary

**Review by rutvikpumak — APPROVED** [review]
Suggest some changes, after that you are good to merge it
Observations :
Code looks well structured and clean
File Structure also looks structured

**Pull request merged**